Sorting/RadixSort.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+    Radix sort
+    ----------
+    An efficient, key-index-based, comparision based sorting algorithm.
+    It sorts based on the keys obtained on each index of the numbers to be sorteed.
+*/
+
+#include <iostream>
+#include <vector>
+#include <array>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+// The setFactors function sets the factors which depend on whether the sorting is needed to be done in ascending or descending order
+
+/*
+    count_sort
+    ----------
+    Performs Count Sort on the elements passed by the radix_sort function
+
+    Time complexity
+    ---------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+
+    Space complexity
+    ----------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    This space is used for count sorting each key and temporarily creating a sorted array.
+*/
+void count_sort(vector<int>& values, const int extractor, const bool to_show_state, const int mult_factor, const int add_factor) {
+    array<int, 10> counter{0};
+    vector<int> output(values.size());
+    for(int i = 0; i < values.size(); i++)
+        counter[add_factor + (mult_factor * ((values[i]/extractor) % 10))]++;
+    for(int i = 1; i < 10; i++)
+        counter[i] += counter[i - 1];
+    for(int i = values.size() - 1; i >= 0; i--) {
+        output[counter[add_factor + (mult_factor * ((values[i]/extractor) % 10))] - 1] = values[i];
+        counter[add_factor + (mult_factor * ((values[i]/extractor) % 10))]--;
+    }
+
+    values = output;
+
+    if(to_show_state) {
+        display_state(values);
+    }
+}
+
+//This function returns the maximum value in the vector
+int getMaxVal(const vector<int> values) {
+    int max_val = values[0];
+    for (int i = 1; i < values.size(); i++) {
```

Comment:
Use [`max_element()`](http://en.cppreference.com/w/cpp/algorithm/max_element) from [`algorithm`](http://en.cppreference.com/w/cpp/algorithm) instead of this function.